src/helm/proxy/clients/openai_client.py

```diff
@@ -187,8 +191,83 @@ def do_it():
             embedding=embedding,
         )
 
+    @staticmethod
+    def _get_tokenizer_name(tokenizer: str) -> str:
+        return tokenizer.split("/")[1]
+
     def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
-        raise NotImplementedError("Use the HuggingFaceClient to tokenize.")
+        # For reproducibility purposes, we use huggingface/gpt2 as the default tokenizer.
+        # except for gpt-3.5 turbo models and gpt-4 that use the tiktoken library.
+
+        if request.tokenizer != "openai/cl100k_base":
+            raise ValueError(
+                f"{request.tokenizer} is not supported."
+                + "Only openai/cl100k-base is supported by the tiktoken library for now."
+            )
+
+        cache_key = asdict(request)
+
+        try:
+
+            def do_it():
+                tokenizer = tiktoken.get_encoding(self._get_tokenizer_name(request.tokenizer))
+                # NOTE(josselin): The tokenizer.encode method returns a list of integers.
+                # This shouuld be fine as TokenizationToken accepts both integers and strings.
+                # In reality, some metrics assume strings, so we decode the integers here.
+                # This is Hacky and should be fixed.
```

Comment:
Could you file a GitHub issue for this and put the GitHub issue number in the comments?

Reply:
Ok so I got to the bottom of this. I was not using `encode` which specifies if the output should be strings or integers. Now everything works fine. Sorry for the confusion.